src/RestWrite.js

```diff
@@ -477,13 +477,14 @@ RestWrite.prototype.validateAuthData = function() {
   );
 };
 
-RestWrite.prototype.handleAuthDataValidation = function(authData) {
+RestWrite.prototype.handleAuthDataValidation = function (authData) {
   const validations = Object.keys(authData).map(provider => {
     if (authData[provider] === null) {
       return Promise.resolve();
     }
     const validateAuthData = this.config.authDataManager.getValidatorForProvider(
-      provider
+      provider,
+      this
```

Comment:
I didn't get it. The idea is to pass the request, right?

Reply:
Yes, I did not find the real right way to test this thing using current tests. 

So here I just added a test that ensure that the object is correctly passed through functions. Do you have any idea to improve this and test it more properly ?

Reply:
Maybe you could do a request using the rest api and sending some custom header and assert that this header is available inside a custom adapter. But I believe it will not work. You are passing the RestWrite instance. That's not the request.